tendrl/notifier/notification/handlers/snmp_handler.py

```diff
@@ -0,0 +1,251 @@
+from etcd import EtcdException
+from etcd import EtcdKeyNotFound
+from pysnmp.error import PySnmpError
+from pysnmp.hlapi import CommunityData
+from pysnmp.hlapi import ContextData
+from pysnmp.hlapi import ObjectIdentity
+from pysnmp.hlapi import ObjectType
+from pysnmp.hlapi import OctetString
+from pysnmp.hlapi import sendNotification
+from pysnmp.hlapi import SnmpEngine
+from pysnmp.hlapi import UdpTransportTarget
+from pysnmp.hlapi import usmDESPrivProtocol
+from pysnmp.hlapi import usmHMACMD5AuthProtocol
+from pysnmp.hlapi import UsmUserData
+from tendrl.commons.config import load_config
+from tendrl.commons.event import Event
+from tendrl.commons.message import ExceptionMessage
+from tendrl.commons.utils.log_utils import log
+from tendrl.notifier.notification import NotificationPlugin
+
+# USM users fixed snmp engine id
+ENGINE_ID = "8000000001020304"
+# default snmp port
+PORT = 162
+V2_ENDPOINT = "v2_endpoint"
+V3_ENDPOINT = "v3_endpoint"
+ALERT_TYPES = {0: "STATUS",
+               1: "UTILIZATION"}
+
+
+class SnmpEndpoint(object):
+    # Base class for SNMP endpoint (credential and host+port pair)
+
+    def __init__(self, host_ip="127.0.0.1", proto=0):
+        self.proto = proto
+        self.host = host_ip
+        self.port = PORT
+        self.engineid = ENGINE_ID
+
+
+class V2Endpoint(SnmpEndpoint):
+    # Class encapsulating an SNMPv2c community, host+port pair
+    # 2 represent SNMPv2
+    def __init__(self, host_ip, community):
+        super(V2Endpoint, self).__init__(host_ip, 2)
+        self.community = community
+
+
+class V3Endpoint(SnmpEndpoint):
+    # Class encapsulating an SNMPv2c community, host+port pair
+    # 2 represent SNMPv2
+    def __init__(self, host_ip, username, auth_key, priv_key):
+        super(V3Endpoint, self).__init__(host_ip, 3)
+        self.usm_user = UsmUserData(userName=username,
+                                    authKey=auth_key,
+                                    privKey=priv_key,
+                                    authProtocol=usmHMACMD5AuthProtocol,
+                                    privProtocol=usmDESPrivProtocol
+                                    )
+
+
+class SnmpHandler(NotificationPlugin):
+
+    def __init__(self):
+        self.name = 'snmp'
+        self.user_configs = []
+
+    def set_destinations(self):
+        try:
+            self.user_configs = self.get_alert_destinations()
+        except (
+            EtcdException,
+            EtcdKeyNotFound,
+            ValueError,
+            KeyError,
+            SyntaxError
+        ) as ex:
+            if type(ex) != EtcdKeyNotFound:
+                self.user_configs = []
+                return
+            raise ex
+
+    def get_alert_destinations(self):
```

Comment:
function name is `get_alert_destinations` and your are returning snmp configurations dict. Looks a bit mismatch.

Reply:
SNMP conf yaml file is having destination details also, snmp conf is not actual configuration  file, we are just maintaining snmp host destination details.

Reply:
Modified var name